lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -520,6 +521,10 @@ tokenize(String, Line, Column, Scope, Tokens) ->
           {error, Reason, String, Tokens};
         _ when Kind == alias ->
           tokenize_alias(Rest, Line, Column, Atom, Length, Ascii, Special, Scope, Tokens);
+        [$= | _] when HasBangOrQuestionMark, Kind == identifier ->
+          Msg = "use a space before = if the previous identifier ends with ! or ?",
+          elixir_errors:warn(Line, Scope#elixir_tokenizer.file, Msg),
+          tokenize_other(Rest, Line, Column, Atom, Length, Scope, Tokens);
         _ when Kind == identifier ->
           tokenize_other(Rest, Line, Column, Atom, Length, Scope, Tokens);
```

Comment:
Why not add a case and check for the last entry only inside this clause then?

Reply:
Basically, let's check for the last only when we are sure we have an identifier.